frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineDetails.tsx

```diff
@@ -3,46 +3,59 @@ import { SectionHeading, ResourceSummary, ResourceLink } from '@console/internal
 import { referenceForModel } from '@console/internal/module/k8s';
 import { Pipeline, getResourceModelFromTask } from '../../../../utils/pipeline-augment';
 import PipelineVisualization from './PipelineVisualization';
+import { usePipelineTriggerTemplateNames } from '../../utils/triggers';
+import { TriggerTemplateModel } from '../../../../models';
+import ResourceLinkList from '../../resource-overview/ResourceLinkList';
 
 interface PipelineDetailsProps {
   obj: Pipeline;
 }
 
-const PipelineDetails: React.FC<PipelineDetailsProps> = ({ obj: pipeline }) => (
-  <div className="co-m-pane__body">
-    <SectionHeading text="Pipeline Details" />
-    <PipelineVisualization pipeline={pipeline} />
-    <div className="row">
-      <div className="col-sm-6">
-        <ResourceSummary resource={pipeline} />
-      </div>
-      {pipeline.spec && pipeline.spec.tasks && (
+const PipelineDetails: React.FC<PipelineDetailsProps> = ({ obj: pipeline }) => {
+  const triggerTemplateNames: string[] = usePipelineTriggerTemplateNames(pipeline) || [];
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text="Pipeline Details" />
+      <PipelineVisualization pipeline={pipeline} />
+      <div className="row">
         <div className="col-sm-6">
-          <SectionHeading text="Tasks" />
-          <dl>
-            {pipeline.spec.tasks.map((task) => {
-              const resourceModel = getResourceModelFromTask(task);
-              return (
-                <React.Fragment key={task.name}>
-                  <dt>Name: {task.name}</dt>
-                  <dd>
-                    Ref:{' '}
-                    <ResourceLink
-                      kind={referenceForModel(resourceModel)}
-                      name={task.taskRef.name}
-                      namespace={pipeline.metadata.namespace}
-                      title={task.taskRef.name}
-                      inline
-                    />
-                  </dd>
-                </React.Fragment>
-              );
-            })}
-          </dl>
+          <ResourceSummary resource={pipeline} />
         </div>
-      )}
+        {pipeline.spec && pipeline.spec.tasks && (
```

Comment:
🤦 Oh right... Pipeline spec is poorly typed. I appreciate you updating this to match types - but `spec` is never not there to have a proper Pipeline and since 0.9 or so of the Operator, you have to have at least 1 task.

Maybe we just update the type of the Pipeline kind and remove this check.